src/trace/log.zig

```diff
@@ -253,114 +240,228 @@ pub const StandardErrLogger = struct {
 /// where some log messages never get logged because the logger is deinitialized before the
 /// logging thread picks up the log message.
 pub const TestLogger = struct {
-    max_level: Level = .warn,
+    const builtin = @import("builtin");
 
     const Self = @This();
-
-    pub const default = TestLogger{};
-
-    pub fn logger(self: Self) Logger {
-        return .{ .test_logger = self };
-    }
-
-    pub fn field(_: Self, _: []const u8, _: anytype) Entry {
-        @panic("`Logger.field` not supported");
-    }
-
-    pub fn info(self: Self, msg: []const u8) void {
-        self.log(.info, msg);
+    max_level: Level,
+    allocator: Allocator,
+    pub fn init(config: Config) *Self {
+        std.debug.assert(builtin.is_test);
+        const self = config.allocator.create(Self) catch @panic("could not allocator.create Logger");
```

Comment:
> It would be nice if the scope that creates the logger would be allowed to make the decision about where the logger itself is allocated 
> ... 
> init could only take max_level as a parameter, and there would be no need for a deinit function

I am not sure I see how that may work, if `init` is only given `max_level` but still need to allocate the logger. Or the idea is that the init returns the logger by value instead of a pointer?

Reply:
>  init returns the logger by value instead of a pointer?

That's the idea. The calling scope can call `Allocator.create` if needed. But I don't think it would actually ever be necessary.

Reply:
I see your point. For the channel logger, it seems fine to return a pointer.

It's not a big deal, just nice to have flexibility when it's low cost.

Reply:
Updated direct print logger to be returned by value in [3d822fc](https://github.com/Syndica/sig/pull/277/commits/3d822fcabab627117c70f6f6a1b81dc8101510f4)

As asides I was wondering what the naming convention is in this case? And if `init` be renamed to `create` since `init` kinda alludes to `deinit` also existing - which is not the case here.

Reply:
I think `init` is fine in both cases. `init` does not imply that `deinit` exists. `init` is just the standard constructor name used in zig. `create` implies that a pointer to an allocated item is being provided. If you were going to use that anywhere, I think it would make more sense to use that for the channel logger because it returns a pointer.